### ISSUE TITLE
Cannot pass a method to BackgroundJob.Enqueue unless it is wrapped in a class local method

### ISSUE THREAD

**Issue by AustinWinstanley** [cady-io/hangfire-ravendb#14]
Doesn't work:

    BackgroundJob.Enqueue(() => System.Console.WriteLine("Background Job: Hello, world!"));

Works:

    BackgroundJob.Enqueue(() => Test());

    [AutomaticRetry(Attempts = 2, LogEvents = true, OnAttemptsExceeded = AttemptsExceededAction.Delete)]
    public static void Test()
    {
        Console.WriteLine("Background Job: Hello, world!"));
    }

**Comment by mjohnson0580** [comment]
I have submitted a pull request #16 that will allow this type of method invocation again.

This error is occurring due to parameter types not being handled in the Job serialization/deserialization and is therefore not allowing overloaded method calls. I have reverted the code back to using InvocationData when serializing Jobs as this functionality and others is already implemented in the InvocationData class.
